pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java

```diff
@@ -469,6 +470,83 @@ public LocalDateTime toLocalDateTime(String s) throws SQLException {
       return result;
     }
   }
+
+  /**
+   * Parse a string and return a LocalDateTime representing its value.
+   *
+   * @param s The ISO formated date string to parse.
+   * @return null if s is null or a LocalDateTime of the parsed string s.
+   * @throws SQLException if there is a problem parsing s.
+   */
+  public OffsetDateTime toOffsetDateTime(String s) throws SQLException {
+    if (s == null) {
+      return null;
+    }
+
+    int slen = s.length();
+
+    // convert postgres's infinity values to internal infinity magic value
+    if (slen == 8 && s.equals("infinity")) {
+      return OffsetDateTime.MAX;
+    }
+
+    if (slen == 9 && s.equals("-infinity")) {
+      return OffsetDateTime.MIN;
+    }
+
+    ParsedTimestamp ts = parseBackendTimestamp(s);
+
+    Calendar tz = ts.tz;
+    int offsetSeconds;
+    if (tz == null) {
+      offsetSeconds = 0;
+    } else {
+      offsetSeconds = tz.get(Calendar.ZONE_OFFSET) / 1000;
+    }
+    ZoneOffset zoneOffset = ZoneOffset.ofTotalSeconds(offsetSeconds);
+    // Postgres is always UTC
+    OffsetDateTime result = OffsetDateTime.of(ts.year, ts.month, ts.day, ts.hour, ts.minute, ts.second, ts.nanos, zoneOffset)
+        .withOffsetSameInstant(ZoneOffset.UTC);
+    if (ts.era == GregorianCalendar.BC) {
+      return result.with(ChronoField.ERA, IsoEra.BCE.getValue());
+    } else {
+      return result;
+    }
+  }
+
+  /**
+   * Returns the offset date time object matching the given bytes with Oid#TIMETZ.
+   *
+   * @param t the time value
+   * @return the matching offset date time
+   */
+  public OffsetDateTime toOffsetDateTime(Time t) {
+    // hardcode utc because the backend does not provide us the timezone
+    // hardoce UNIX epoch, JDBC requires OffsetDateTime but doesn't describe what date should be used
+    return t.toLocalTime().atDate(LocalDate.of(1970, 1, 1)).atOffset(ZoneOffset.UTC);
+  }
+
+  /**
+   * Returns the offset date time object matching the given bytes with Oid#TIMESTAMPTZ.
+   *
+   * @param bytes The binary encoded local date time value.
+   * @return The parsed local date time object.
+   * @throws PSQLException If binary format could not be parsed.
+   */
+  public OffsetDateTime toOffsetDateTimeBin(byte[] bytes) throws PSQLException {
+    ParsedBinaryTimestamp parsedTimestamp = this.toProlepticParsedTimestampBin(bytes);
+    if (parsedTimestamp.infinity == Infinity.POSITIVE) {
```

Comment:
Everywhere else we use PGStatement.DATE_POSITIVE_INFINITY I think we have to stay consistent unless there is something I am missing ?

Reply:
The existing code already uses the .MAX fields in case of infinity. So using the .MAX fields keeps the existing behavior of the driver.

* https://github.com/pgjdbc/pgjdbc/blob/0600990007669119b73ee2adb064184a4c62343f/pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java#L454
* https://github.com/pgjdbc/pgjdbc/blob/0600990007669119b73ee2adb064184a4c62343f/pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java#L1137
* https://github.com/pgjdbc/pgjdbc/blob/0600990007669119b73ee2adb064184a4c62343f/pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java#L3337

`org.postgresql.PGStatement#DATE_POSITIVE_INFINITY` is currently at `292278994-08-16 13:00:00Z AD` and `DATE_NEGATIVE_INFINITY` is currently as `292269055-12-02 13:00:00Z BC`. These are driven by the implementation of the `java.util.Date` (and implicitly `java.sql.Timestamp`) classes being a single `long`. The `java.time` types can to as hight as 999,999,999 AD and 999,999,999 BC. As the `java.time` types where developped specifically to overcome the limitation of the `java.util.Date` and `java.sql` types I don't think it is a good idea to impose the restrictions of the old types on the new types.